### DISCUSSION TITLE
GitHub Classroom & GitHub Desktop - Error on fetch

### DISCUSSION THREAD

**Discussion by jrshelby** [community/community#136955] [GitHub Education]
### Hi everyone,

We're experiencing a weird side effect of the switch to GitHub Classroom assignments forking.

Currently, I am:
- Creating a private template repo with starter code.
- Creating a Classroom assignment, which makes a private copy of that template.
- Students accept the assignment and get their private fork of the copy.

For individual assignments, everything is private here (both the template and student forks). Part of this is because I'd like to keep solutions as a branch within the template. Another is so that students are not confused by available repos - they only see their own repos in the class organization.

The problem:
We use GitHub Desktop to work with our repos. When GitHub classroom fetches changes, students are getting an error on every fetch that reads "the repository does not seem to exist anymore". On closer inspection, the fetch is working just fine, and if there are commits on their repo, they'll be able to pull them.

We're getting the error because **the fetch is trying to access both the student repo and the forked parent repo**. The students cannot access the parent repo because it is private. Thus, an error every time they fetch.

After some Googling, it sounds like
- It is standard behavior of GitHub Desktop to fetch the parent repo.
- It is impossible to remove the upstream parent.

A bunch of posts about this issue on the GitHub Desktop forums suggest that is in non-ideal, but is probably a rare edge case for GitHub Desktop use. I wanted to highlight the issue here because **the new behavior of GitHub Classroom makes this not an edge case anymore**. Probably a lot of educators have a similar set up to myself.

If anyone knows a workaround for this, I'd be interested! It's not the end of the world - the fetch still goes through - but I work with high school students and it'd be great to simplify this process.

**Comment by wfoxcode** [comment]
Hello,
I wanted to second this, is there any comment/update that can come from the github reps? Currently it is impossible for students to clone their classroom assignment repo's, which makes for an unideal situation when utilizing desktop IDE's. 

If anyone has figured out a work around, I would love to hear.

**Comment by jrshelby** [comment]
By the way, for reference, here is the corresponding issue posted on GitHub Desktop:
https://github.com/desktop/desktop/issues/3715

**Comment by crashbash111** [comment]
Commenting to indicate that we've got the same issue since GitHub Classroom moved to forked repositories. Definitely seems to be tied to the upstream repository being privated automatically. Have not been able to find a solution via GitHub Desktop outside of manually marking the upstream repository as public (which is not ideal).

**Comment by angelhsu05** [comment]
I also experienced this issue for the first time. Some students showed a new option when they go to sync their assignment repo using GitHub Desktop where it asks whether they want to contribute code or just use it for personal use. It seems the students who selected 'for my personal use' didn't have this problem. I'm going to have the students who didn't select this option to try this and will report back.

**Reply by wfoxcode** [reply]
> I also experienced this issue for the first time. Some students showed a new option when they go to sync their assignment repo using GitHub Desktop where it asks whether they want to contribute code or just use it for personal use. It seems the students who selected 'for my personal use' didn't have this problem. I'm going to have the students who didn't select this option to try this and will report back.

That is an interesting note, I will have to check this out this upcoming semester. I just opted to not use classroom at all this Fall due to this limitation, it would be nice if this resolves some of the problems.

**Comment by angelhsu05** [comment]
Yes in GitHub desktop there’s also an option in Repository settings to select the Forking options to just use for “personal use.” That seems to have resolved the issues.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by wgmitchener** [reply]
I tried this and GitHub Desktop still tries to pull from the private upstream remote. I've resorted to making the template repository public. It's also possible to remove the upstream remote with git at the command line, which complicates students pulling an update to that upstream repository. Definitely not a corner case any more.
